src/js/autocomplete.js

```diff
@@ -6,9 +13,187 @@ class Autocomplete {
 	 */
 	constructor (autocompleteEl, options) {
 		this.autocompleteEl = autocompleteEl;
-		this.options = Object.assign({}, {
+		this.options = Object.assign({
+			placeholder: '',
+			cssNamespace: 'o-autocomplete',
+			displayMenu: 'overlay',
+			showNoOptionsFound: false,
+			templates: {
+				suggestion: this.suggestionTemplate.bind(this)
+			}
 		}, options || Autocomplete.getDataAttributes(autocompleteEl));
+
+		// If highlighter is a string, then it is the name of a global function to use.
+		// If highlighter is not a string, then it is a function to use.
+		if (typeof this.options.highlighter === 'string') {
+			this.options.highlighter = window[this.options.highlighter];
+		}
+
+		const container = document.createElement('div');
+		this.container = container;
+		container.classList.add('o-autocomplete__container');
+
+		if (this.options.source) {
+			// If source is a string, then it is the name of a global function to use.
+			// If source is not a string, then it is a function to use.
+			/**
+			 * @function
+			 * @param {string} query - Text which was typed into the autocomplete by the user
+			 * @param {PopulateResults} populateResults - Function to call when ready to update the suggestions dropdown
+			 * @returns {void}
+			 */
+			const customSuggestions = typeof this.options.source === 'string' ? window[this.options.source] : this.options.source;
```

Comment:
TODO: Need to write a test for the source function

Reply:
Done in [`2dfd0d2` (#2)](https://github.com/Financial-Times/o-autocomplete/pull/2/commits/2dfd0d273e3de893f4f45931b44f1a704cdc7fa4)